libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/alert/RouteAlertGeometry.kt

```diff
@@ -23,25 +23,14 @@ import com.mapbox.geojson.Point
  * @param endCoordinate point where the alert ends.
  * @param endGeometryIndex index of a point in the route geometry where the alert ends.
  */
-class RouteAlertGeometry private constructor(
+class RouteAlertGeometry internal constructor(
     val length: Double,
     val startCoordinate: Point,
     val startGeometryIndex: Int,
     val endCoordinate: Point,
     val endGeometryIndex: Int
 ) {
 
-    /**
-     * Transform this object into a builder to mutate the values.
-     */
-    fun toBuilder() = Builder(
```

Comment:
Removing the builders, we end up losing the things we were building.

1. no longer have consistent domain models
1. no longer can use the [BuilderTest](https://github.com/mapbox/mapbox-navigation-android/blob/master/libtesting-utils/src/main/java/com/mapbox/navigation/testing/BuilderTest.kt), losing easy test coverage
1. re-introduce constructor ambiguity (Double, Point, Int, Point, Int)
1. lose the nice Java 
...

There were reasons we wanted to use builders, maybe we don't. Maybe we should move these models to mapbox-java?

IMO we should have these discussions separate from regular feature development. Is this feature introducing something we haven't seen anywhere else, and now we want to change our domain pattern?

Reply:
But then again, we also would like to separate caller vs callee

For example, RouteProgress does not need a builder because it comes from the SDK. The only reason RouteLegProgress has a builder, is because of our logic here 
https://github.com/mapbox/mapbox-navigation-android/blob/e2b04a921c6e2d71d55e7ac5261eeafced6fbd26/libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/NavigatorMapper.kt#L59-L61

The RouteAlertGeometry is a domain object that is always coming from the SDK. If that is the motivation behind removing the builders, it would be good to be more explicit with this. I wanted to delete the RouteLegProgress.Builder too. It may be easier now that we can test it in NavigatorMapper